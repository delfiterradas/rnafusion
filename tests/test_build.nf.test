nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    profile "test_build"
    tag "pipeline"
    tag "pipeline_rnafusion"
    tag "test_build"

    test("Test build references") {
        when {
            params {
                config_profile_name        = 'Test build references profile'
                config_profile_description = 'Minimal test dataset to check pipeline function'

                // Input data
                references_only            = true
                input                      = "${projectDir}/tests/csv/fastq.csv"
                genomes_base               = "${outputDir}/references"
                arriba_ref_protein_domains = "https://github.com/nf-core/test-datasets/raw/refs/heads/rnafusion/testdata/reference/arriba/protein_domains_hg38_GRCh38_v2.5.0.gff3"
                arriba_ref_known_fusions   = "https://github.com/nf-core/test-datasets/raw/refs/heads/rnafusion/testdata/reference/arriba/known_fusions_hg38_GRCh38_v2.5.0.tsv.gz"
                arriba_ref_blacklist       = "https://github.com/nf-core/test-datasets/raw/refs/heads/rnafusion/testdata/reference/arriba/blacklist_hg38_GRCh38_v2.5.0.tsv.gz"
                arriba_ref_cytobands       = "https://github.com/nf-core/test-datasets/raw/refs/heads/rnafusion/testdata/reference/arriba/cytobands_hg38_GRCh38_v2.5.0.tsv"
                fusioncatcher_ref          = "${projectDir}/assets" // This is a placeholder until a small reference can be created
                no_cosmic                  = true
                tools                      = "all"
                fasta                      = 'https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/minigenome.fa'
                gtf                        = 'https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/minigenome.gtf'
                fusion_annot_lib           = 'https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/CTAT_HumanFusionLib.mini.dat.gz'
                species                    = 'homo_sapiens'
                dfam_version               = 3.4
                outdir                     = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
