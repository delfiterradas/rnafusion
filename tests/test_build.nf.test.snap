{
    "test_build": {
        "content": [
            [
                "bedops",
                "bedops/Homo_sapiens.GRCh38.102_rrna_intervals.gtf.bed",
                "gatk4",
                "gatk4/Homo_sapiens.GRCh38.102.dna.primary_assembly.dict",
                "gffread",
                "gffread/Homo_sapiens.GRCh38.102.fasta",
                "pipeline_info",
                "references",
                "references/ensembl",
                "references/ensembl/Homo_sapiens.GRCh38.102.chr.gtf",
                "references/ensembl/Homo_sapiens.GRCh38.102.dna.primary_assembly.fa",
                "references/ensembl/Homo_sapiens.GRCh38.102.dna.primary_assembly.fa.fai",
                "references/ensembl/Homo_sapiens.GRCh38.102.genepred",
                "references/ensembl/Homo_sapiens.GRCh38.102.gtf",
                "references/ensembl/Homo_sapiens.GRCh38.102.refflat",
                "references/ensembl/Homo_sapiens.GRCh38.102_rrna_intervals.gtf.interval_list",
                "references/hgnc",
                "references/hgnc/HGNC-DB-timestamp.txt",
                "references/hgnc/hgnc_complete_set.txt",
                "rrnatranscripts",
                "rrnatranscripts/Homo_sapiens.GRCh38.102_rrna_intervals.gtf"
            ],
            [
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf.bed:md5,5bc3ccc76735ae46699a75269f0ea65b",
                "Homo_sapiens.GRCh38.102.dna.primary_assembly.dict:md5,092f1bf29fce906ff7b5ece02b4b21c8",
                "Homo_sapiens.GRCh38.102.fasta:md5,d1f17b045dc60c49f2cc29e30006afc0",
                "Homo_sapiens.GRCh38.102.chr.gtf:md5,0069687307852c63596b4d2ebdbbaf0c",
                "Homo_sapiens.GRCh38.102.dna.primary_assembly.fa.fai:md5,d527f3eb6b664020cf4d882b5820056f",
                "Homo_sapiens.GRCh38.102.genepred:md5,59c577ca4ab033c0ce1feac1e387bcab",
                "Homo_sapiens.GRCh38.102.gtf:md5,defac755cd9aa4e82ec33398c27745ef",
                "Homo_sapiens.GRCh38.102.refflat:md5,ef095e13743811c31d44752c32e9673e",
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf.interval_list:md5,0abf61877f65247b15c438d605d85599",
                "hgnc_complete_set.txt:md5,a563a2f8432ec0ab7d3dc74d769102b8",
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf:md5,744bf505deb50837b15441e808cad345"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-09T13:33:28.517098377"
    }
}